src/main/java/com/aws/greengrass/clientdevices/auth/iot/CertificateRegistry.java

```diff
@@ -55,7 +64,7 @@ public CertificateRegistry(RuntimeConfiguration runtimeConfiguration, ClientCert
     public Optional<Certificate> getCertificateFromPem(String certificatePem) throws InvalidCertificateException {
         Certificate cert = Certificate.fromPem(certificatePem);
         Optional<CertificateV1DTO> dto = runtimeConfiguration.getCertificateV1(cert.getCertificateId());
-        return dto.map(this::certificateV1DTOToCert);
+        return dto.map(this::certificateV1DTOToCert).filter(this::isTrusted);
```

Comment:
This is just filtering out what we don't trust anymore but not really changing the value we have stored. I don't think the registry is the place that should decide wether something is trusted or not. There are a few options I can think of.

1) We have a some sort of certificate factory that knows about this setting and when creating a certificate it assigns the TLL to the certificate it is returning. Then the certificate itself would have a method called `certificate.isValid` (this one is preferred)

2) We have something that tracks periodically checks for devices and their TTL and marks them as expired when they are no longer active.

Reply:
went with #1, thanks